source/lac/petsc_precondition.cc

```diff
@@ -94,28 +81,39 @@ namespace PETScWrappers
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
 
+  void
+  PreconditionBase::setup()
+  {
+    AssertThrow(pc != nullptr, StandardExceptions::ExcInvalidState());
 
+    PetscErrorCode ierr = PCSetUp(pc);
+    AssertThrow(ierr == 0, ExcPETScError(ierr));
+  }
 
-  MPI_Comm
+  const MPI_Comm &
   PreconditionBase::get_mpi_communicator() const
   {
-    return mpi_communicator;
+    static MPI_Comm comm  = PETSC_COMM_SELF;
+    MPI_Comm        pcomm = PetscObjectComm(reinterpret_cast<PetscObject>(pc));
+    if (pcomm != MPI_COMM_NULL)
+      comm = pcomm;
+    return comm;
   }
```

Comment:
Wasn't the consensus that we want to move away from returning references?

(I'm ok with this here, if the plan is to change it back soon.)

Reply:
Yes, and here you have the commit that unifies those getters https://github.com/dealii/dealii/pull/14481/commits/0db14d853d927f7dbb45918d71ebbc0c33dd10a9